packages/clerk-js/src/core/devBrowserHandler.ts

```diff
@@ -107,17 +106,19 @@ export default function createDevBrowserHandler({
 
   async function setUrlBasedSessionSyncBrowser(): Promise<void> {
     // 1. Get the JWT from hash search parameters when the redirection comes from Clerk Hosted Pages
-    const devBrowserToken = getSearchParameterFromHash({
-      hash: window.location.hash,
-      paramName: DEV_BROWSER_SSO_JWT_PARAMETER,
-    });
+    let preExistingHash = true;
+    let arr = window.location.hash.split(DEV_BROWSER_SSO_JWT_PARAMETER_WITH_HASH_SEPARATOR);
+    if (arr.length === 1) {
+      preExistingHash = false;
+      arr = window.location.hash.split(DEV_BROWSER_SSO_JWT_PARAMETER_SEPARATOR);
+    }
+    const devBrowserToken = arr.length > 1 ? arr[1] : undefined;
 
     if (devBrowserToken) {
-      window.location.hash = removeSearchParameterFromHash({
-        hash: window.location.hash,
-        paramName: DEV_BROWSER_SSO_JWT_PARAMETER,
-      });
-
+      window.location.hash = window.location.hash.split(DEV_BROWSER_SSO_JWT_PARAMETER_SEPARATOR)[0];
```

Comment:
❓ Is there any cleaner way to write this? Accessing array positions directly feels a little confusing.

Reply:
We're just getting the first part of the hash without the JWT in this case. I think it's a little overkill to make a util for this though.